### PR TITLE
Add UPS support for other labels like ZPL, EPL STARPL and SPL

### PR DESCRIPTION
**Pull request by wassimk**
This fixes and closes PR #316 too.

### CODE REVIEW

File: lib/active_shipping/carriers/ups.rb [hunk 1]
```diff
@@ -450,15 +450,26 @@ def build_shipment_request(origin, destination, packages, options={})
             end
           end
 
-          # I don't know all of the options that UPS supports for labels
-          # so I'm going with something very simple for now.
+          # Supported image_formats:
+          # GIF, EPL, ZPL, STARPL and SPL
+          label_format = options[:label_format].nil? ? 'GIF' : options[:label_format].upcase
```
Comment (kmcphillips) [inline lib/active_shipping/carriers/ups.rb]: This still allows for `options[:label_format] = ""`

File: lib/active_shipping/carriers/ups.rb [hunk 2]
```diff
@@ -450,15 +450,26 @@ def build_shipment_request(origin, destination, packages, options={})
...
+
             xml.LabelPrintMethod do
-              xml.Code('GIF')
+              xml.Code(label_format)
             end
-            xml.HTTPUserAgent('Mozilla/4.5') # hmmm
-            xml.LabelImageFormat('GIF') do
-              xml.Code('GIF')
+
+            xml.HTTPUserAgent('Mozilla/4.5')
```
Comment (kmcphillips) [inline lib/active_shipping/carriers/ups.rb]: What's this about?

Comment (eboswort) [inline lib/active_shipping/carriers/ups.rb]: In the UPS Api docs the description of this field is `Browser HTTPUserAgent String. This is the preferred way of identifying GIF image type to be generated`. They say it is `Required if ShipmentConfirmRequest/LabelSpecification/LabelPrintMethod/Code = GIF. Default to Mozilla/4.5 if this field is missing or has invalid value`

Comment (eboswort) [inline lib/active_shipping/carriers/ups.rb]: Might be best to put this inside of the `if label_format == 'GIF'` block with a comment about how it's a stupid thing that UPS requires

File: lib/active_shipping/carriers/ups.rb (hunk 1 above)
Comment (wassimk) [inline lib/active_shipping/carriers/ups.rb]: I just reformatted it. If you don't specify a label format the API will return appropriate error. How does that look?

File: lib/active_shipping/carriers/ups.rb (hunk 2 above)
Comment (kmcphillips) [inline lib/active_shipping/carriers/ups.rb]: +1

File: lib/active_shipping/carriers/ups.rb (hunk 1 above)
Comment (kmcphillips) [inline lib/active_shipping/carriers/ups.rb]: :+1:

**Comment by tmecklem** [comment]
I also need this support for epl image formats. I'll pitch in if anything else needs to be done to accept this PR.

**Comment by kmcphillips** [comment]
Can you rebase, get the conflicts fixed, and the build green?

**Comment by wassimk** [comment]
Hey @kmcphillips please let me know if we need to add anything else to merge this in.

**Comment by kmcphillips** [comment]
Thanks for the good lookin' tests

**Comment by kmcphillips** [comment]
Released in 1.6.3

**Comment by tmecklem** [comment]
It looks like the height and width for labelstocksize might be backwards. I'm seeing height of 4 and width of 6 or 8, but the API docs say that valid values for width is 4 and height is 6 or 8.

**Comment by tmecklem** [comment]
On trying out "correct" values for height and width, the api rejects the request. This seems to be a bug on UPS' side.

**Comment by eboswort** [comment]
I've noticed the label images that come back from UPS are sideways. Maybe these two things are related.

**Pull request merged**